### ISSUE TITLE
ConversationalChatAgent sometimes outputs multiple actions in response

### ISSUE THREAD

**Issue by jbbakst** [langchain-ai/langchain#2567]
When using the `ConversationalChatAgent`, it sometimes outputs multiple actions in a single response, causing the following error:

```
ValueError: Could not parse LLM output: ...
```

Ideas:
1. Support this behavior so a single `AgentExecutor` run loop can perform multiple actions
2. Adjust prompting strategy to prevent this from happening

On the second point, I've found that explicitly adding \`\`\`json as an `AIMessage` in the `agent_scratchpad` and then handling that in the output parser seems to reliably lead to outputs with only a single action. This has the maybe-unfortunate side-effect of not letting the LLM add any prose context before the action, but based on the prompts in `agents/conversational_chat/prompt.py`, it seems like that's already not intended. E.g. this seems to help with the issue:

```
class MyOutputParser(AgentOutputParser):
    def parse(self, text: str) -> Any:
        # Add ```json back to the text, since we manually added it as an AIMessage in create_prompt
        return super().parse(f"```json{text}")


class MyAgent(ConversationalChatAgent):
    def _construct_scratchpad(
        self, intermediate_steps: List[Tuple[AgentAction, str]]
    ) -> List[BaseMessage]:
        thoughts = super()._construct_scratchpad(intermediate_steps)
        # Manually append an AIMessage with ```json to better guide the LLM towards responding with only one action and no prose.
        thoughts.append(AIMessage(content="```json"))
        return thoughts

    @classmethod
    def create_prompt(
        cls,
        tools: Sequence[BaseTool],
        system_message: str = PREFIX,
        human_message: str = SUFFIX,
        input_variables: Optional[List[str]] = None,
        output_parser: Optional[BaseOutputParser] = None,
    ) -> BasePromptTemplate:
        return super().create_prompt(
            tools,
            system_message,
            human_message,
            input_variables,
            output_parser or MyOutputParser(),
        )

    @classmethod
    def from_llm_and_tools(
        cls,
        llm: BaseLanguageModel,
        tools: Sequence[BaseTool],
        callback_manager: Optional[BaseCallbackManager] = None,
        system_message: str = PREFIX,
        human_message: str = SUFFIX,
        input_variables: Optional[List[str]] = None,
        output_parser: Optional[BaseOutputParser] = None,
        **kwargs: Any,
    ) -> Agent:
        return super().from_llm_and_tools(
            llm,
            tools,
            callback_manager,
            system_message,
            human_message,
            input_variables,
            output_parser or MyOutputParser(),
            **kwargs,
        )
```

**Comment by aeturneus** [comment]
Seeing the same issue

**Comment by treppers** [comment]
Same.

My work around was just to parse the first action listed and basically ignore the second action.  Still working to see how that impacts the agents performance - so this is far from a solution.

If anyone cares to know what I did, I created a custom output parser (basically copy pasta the default one) but add `maxsplit=1` to the end of the split function.

Udpated snippet looks like this:
`        if "```json" in cleaned_output:
            _, cleaned_output = cleaned_output.split("```json", maxsplit=1)
        if "```" in cleaned_output:
            cleaned_output, _ = cleaned_output.split("```", maxsplit=1)`
